sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala

```diff
@@ -194,12 +203,28 @@ class V2SessionCatalog(catalog: SessionCatalog)
   }
 
   override def dropTable(ident: Identifier): Boolean = {
+    dropTableInternal(ident)
+  }
+
+  private def dropTableInternal(ident: Identifier, purge: Boolean = false): Boolean = {
     try {
-      if (loadTable(ident) != null) {
+      val table = loadTable(ident)
+      if (table != null) {
+        val v1Table = table.asInstanceOf[V1Table].v1Table
```

Comment:
people can provide a custom v2 session and `loadTable` does not always return a v1 table. Can we do a pattern match here and only do the additional handling if it's a v1 table?

Reply:
Fixed.